### PR TITLE
This commit fixes several issues with split-archive creation

### PR DESCRIPTION
**Pull request by akahles**
* modified the splitter, such that not only the splits, but a complete file listing (including empty directories and recursively parsed directories)
* streamlined archiver.py to make the processing of split and unsplit archives more alike (first step towards this, integration can go further)
* archive creation now uses the complete file list and not the checksum file
* the additional *.lst files have been added to all relevant tests
* symlinks to directories are now treated as files
* creation of tar archives now uses the option "--no-recursion”
* renamed "create_filelist_and_hashs” into “create_filelist_and_hashes"

### CODE REVIEW

File: archiver/helpers.py [hunk 1]
```diff
@@ -140,14 +157,25 @@ def hash_files_and_check_symlinks(source_path, abs_paths, max_workers=1, integri
...
+        dirs_ = []
+        for dir in dirs:
+            abs_dir = root_path.joinpath(dir)
+            if Path(abs_dir).is_symlink():
+                listing.append(abs_dir)
+            else:
+                if include_dirs:
+                    listing.append(abs_dir)
+                dirs_.append(dir)
+        dirs = dirs_
```
Comment (izcram) [inline archiver/helpers.py:176]: This should be `dirs[:] = dirs_` to properly mutate the list and not just reassigning to a new list, which would be ignored by os.walk

**Review by izcram — CHANGES_REQUESTED** [review]
There is a small potential bug in get_files_in_folder, otherwise LGTM. Nice, that symlink folders  and empty folders are properly handled. 
Maybe it would be good to include a new testcase with a symlinked folder and empty folder.
At a later staged, the integrity test could be extended to also use the file listings to detect missing empty folders (perhaps not that urgent)

File: archiver/splitter.py [hunk 2]
```diff
@@ -15,15 +16,25 @@ def split_directory(directory_path, max_package_size):
...
 
             if archive_size + dir_size < max_package_size:
                 current_archive.append(dir_path)
+                current_listing.append(dir_path)
+                current_listing.extend(helpers.get_files_in_folder(dir_path, include_dirs=True))
                 archive_size += dir_size
 
                 excluded_dirs.append(directory)
+            else:
+                current_listing.append(dir_path)
```
[the remark this answers is not in the document]
Comment (akahles) [reply archiver/splitter.py:36]: Makes sense.  Done.

File: archiver/helpers.py (hunk 1 above)
Comment (akahles) [reply archiver/helpers.py:176]: Ah, good catch. I assume my tests have been too simple :) Will improve

**Review by izcram — APPROVED** [review]

**Pull request merged**